### PR TITLE
Name snapshots consistently

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

Snapshot names are overloaded, and there are multiple terms that mean the same thing. This is confusing. Here's a list of ones in the codebase that I've found:

```
- snapshot_dir
- snapshots_dir
- snapshot_path
- snapshot_output_dir
- snapshot_package_output_path
- snapshot_archives_dir
```

#### Summary of Changes

For all the ones that are about the directory where snapshot archives are stored, ensure they are `snapshot_archives_dir`. For the ones about the (bank) snapshots directory, set to `bank_snapshots_dir`.

### CODE REVIEW

File: core/tests/snapshots.rs [hunk 1]
```diff
@@ -273,15 +273,16 @@ mod tests {
         // Generate a snapshot package for last bank
         let last_bank = bank_forks.get(last_slot).unwrap();
         let snapshot_config = &snapshot_test_config.snapshot_config;
-        let snapshot_path = &snapshot_config.snapshot_path;
-        let last_bank_snapshot_info = snapshot_utils::get_highest_bank_snapshot_info(snapshot_path)
-            .expect("no snapshots found in path");
+        let bank_snapshots_dir = &snapshot_config.bank_snapshots_dir;
+        let last_bank_snapshot_info =
+            snapshot_utils::get_highest_bank_snapshot_info(bank_snapshots_dir)
+                .expect("no snapshots found in path");
```
Comment (mvines) [inline core/tests/snapshots.rs:279]:
```suggestion
                .expect("no bank snapshots found in path");
```

File: core/tests/snapshots.rs [hunk 2]
```diff
@@ -754,15 +754,18 @@ mod tests {
...
-        let bank_snapshot_info = snapshot_utils::get_bank_snapshots(&snapshot_config.snapshot_path)
-            .into_iter()
-            .find(|elem| elem.slot == slot)
-            .ok_or_else(|| Error::new(ErrorKind::Other, "did not find snapshot with this path"))?;
+        let bank_snapshot_info =
+            snapshot_utils::get_bank_snapshots(&snapshot_config.bank_snapshots_dir)
+                .into_iter()
+                .find(|elem| elem.slot == slot)
+                .ok_or_else(|| {
+                    Error::new(ErrorKind::Other, "did not find snapshot with this path")
```
Comment (mvines) [inline core/tests/snapshots.rs:762]:
```suggestion
                    Error::new(ErrorKind::Other, "did not find bank snapshot with this path")
```

File: core/tests/snapshots.rs [hunk 3]
```diff
@@ -783,10 +786,13 @@ mod tests {
...
```
Comment (mvines) [inline core/tests/snapshots.rs:794]:
```suggestion
                    Error::new(ErrorKind::Other, "did not find bank snapshot with this path")
```

File: ledger/src/bank_forks_utils.rs [hunk 4]
```diff
@@ -43,14 +43,14 @@ pub fn load(
     if let Some(snapshot_config) = snapshot_config {
         info!(
             "Initializing snapshot path: {}",
```
Comment (mvines) [inline ledger/src/bank_forks_utils.rs:45]:
```suggestion
            "Initializing bank snapshot path: {}",
```

File: replica-node/src/replica_node.rs [hunk 5]
```diff
@@ -82,24 +82,26 @@ fn initialize_from_snapshot(
...
         replica_config.snapshot_info,
         false,
         snapshot_config.maximum_snapshots_to_retain,
         &mut None,
     )
     .unwrap();
 
-    fs::create_dir_all(&snapshot_config.snapshot_path).expect("Couldn't create snapshot directory");
+    fs::create_dir_all(&snapshot_config.bank_snapshots_dir)
+        .expect("Couldn't create snapshot directory");
```
Comment (mvines) [inline replica-node/src/replica_node.rs:99]:
```suggestion
        .expect("Couldn't create bank snapshot directory");
```

**Review by mvines — DISMISSED** [review]
Thanks for taking the time to improve code quality like this!

**Review by CriesofCarrots — APPROVED** [review]
r+ mvines's suggestions
Thanks for the cleanup!

**Pull request merged**